app/assets/javascripts/components/expander.js

```diff
@@ -69,17 +69,21 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}; // if this ; is omitted, none
     $button.setAttribute('aria-expanded', expanded)
     $button.setAttribute('aria-controls', this.$content.getAttribute('id'))
 
-    // GA4 Accordion tracking. Relies on the ga4-finder-tracker setting the index first, so we wrap this in a custom event.
-    window.addEventListener('ga4-filter-indexes-added', function () {
-      if (window.GOVUK.analyticsGa4) {
-        if (window.GOVUK.analyticsGa4.Ga4FinderTracker) {
-          window.GOVUK.analyticsGa4.Ga4FinderTracker.addFilterButtonTracking($button, this.$toggle.innerHTML)
+    var buttonAttributes = this.$module.getAttribute('data-button-data-attributes')
+    if (buttonAttributes) {
+      try {
+        buttonAttributes = JSON.parse(buttonAttributes)
+        for (var rawKey in buttonAttributes) {
+          var key = rawKey.replace(/_/i, '-').toLowerCase()
```

Comment:
I just noticed something with this regex which also applies to the one in `govuk_publishing_components`

Would `/i` be irrelevant here as it's capturing a symbol rather than a character, so setting it to insensitive isn't necessary?

Also, should it have the `/g` flag instead so that it replaces every instance of an underscore in the key? With the current regex, I think it only works if the key only has one underscore?

Reply:
That's really helpful, thanks @AshGDS. Yes I think you're right, although for readability I've gone with the `replaceAll` function. I've also updated the tests to reflect this.

Reply:
Once I'm happy with this I'll make the same change on the option select component in the gem as well.

Reply:
Thanks @andysellick - just to confirm, are we dropping IE11 support? If I recall correctly `replaceAll` doesn't work in IE11 (and it seems to not be working when I check in IE11 via BrowserStack). If we're not supporting IE11 anymore then happy to approve the change :+1:

Reply:
Ah thanks for spotting that, hadn't realised. Have reverted it back to `replace`.